documentation/course/e2/README.md

```diff
@@ -1436,7 +1437,140 @@ Model danych serwisu jest na tyle prosty, a jednocześnie serwis tak uniwersalni
 
 ### Bilet
 
-TODO @jakubzehner: Dodać diagram bazodanowy do Clobbert, dodać uzasadnienia dla decyzji i uzupełnić tabelę.
+Model informacyjny podsystemu `clabbert` składa się z sześciu klas i dwóch enumeratorów. Cztery spośród tych klas tworzą hierarchię dziedziczenia oferty biletowej. Abstrakcyjna klasa `TicketOffer` zawiera wspólne elementy dla każdego z typów oferty biletowej, natomiast klasy `TimeLimitedOffer`, `SingleFareOffer` oraz `LongTermOffer` reprezentują dostępne rodzaje oferty biletowej i za wyjątkiem `SingleFareOffer` wprowadzają nowe dane. Oddzielny sposób reprezentacji klientowi informacji o różnych ofertach oraz wymagane w implementacji mechanizmy unikalne dla danego rodzaju oferty spowodowały, że zdecydowano się na zamodelowanie bazy danych w postaci podejścia _table-per-type_, czyli tabeli dla każdej z klas dziedziczących, łączonych z tabelą dla abstrakcyjnej klasy bazowej. Dodatkowo zdecydowana się na reprezentację enumeratorów w postaci tekstu, konkretniej `varchar(32)`, ze względu na większą czytelność danych kolumny w tabeli i braku ewentualnych problemów podczas dodania kolejnej wartości enumeratora na miejscu innym niż ostatnim. Podejście te jednak poświęca odrobinę wydajności, względem drugiego najpopularniejszego rozwiązania, czyli zastosowania liczby porządkowej odpowiadającej wartości enumeratora.
```

Comment:
> Dodatkowo zdecydowan*a* się na reprezentację

typo

Instead of doing `varchar` for enumerator maybe use [`ENUM`](https://www.postgresql.org/docs/current/datatype-enum.html). iirc Postgres automagicaly formats them for viewing as its text counterpart. Also `ENUMs` can be extended.